cranelift/codegen/src/isa/x64/lower.isle

```diff
@@ -2912,6 +2912,11 @@
       (side_effect
        (x64_movrm $I32 (to_amode flags address offset) value)))
 
+;; IMM stores
+(rule 2 (lower (store flags (has_type (fits_in_64 ty) (iconst (u64_from_imm64 value))) address offset))
+      (side_effect
+       (x64_movimm_m ty (to_amode flags address offset) value)))
```

Comment:
If you're up for it I think it might be good to update the `MovImmM` variant itself. Looking at `emit.rs` it will panic if the low 32-bits of the `value` here don't sign-extend into the full 64-bits, so the `MovImmM` should I think store `i32` instead of `i64`. With that you could add extra matching here that `value` fits within a 32-bit signed integer.